### PR TITLE
Fixed #609: Added app for board meeting minutes.

### LINKED ISSUE #609: Create a model to track foundation meeting minutes
Currently all board meeting minute pages are flatpages.

This idea is to create a "foundation" app with a `BoardMinutes` model that has a few fields: date (of meeting), body, content_format (html or restructured text, like the `Entry` model in the `blog` app).

Then add a few views to automate generation of the minutes pages (e.g. /foundation/records/minutes/2015-08-06/) and the index (/foundation/records/minutes/) instead of using flatpages. We'll link to the new index page from https://www.djangoproject.com/foundation/records/.

### PR DESCRIPTION
**Pull request by ubernostrum**
This implements models, views and templates for tracking minutes of DSF Board meetings, based on the structure of the (private) documents maintained by the Secretary (currently, me).

This does *not* import any historical minutes into the new structure; that's left as a followup, since right now the more important thing is making it easy to post the current minutes. Once merged, this will require some changes to existing DSF flat pages to link the archive of the new minutes app, but I can handle that once the app is deployed and the backlog of unposted minutes is cleared.

### CODE REVIEW

File: foundation/views.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+
+class MeetingArchiveDay(MeetingMixin, generic.DayArchiveView):
+    pass
+
+
+class MeetingDetail(MeetingMixin, generic.DateDetailView):
+    context_object_name = 'meeting'
+
+    def get_querysett(self):
```
Comment (adamchainz) [inline foundation/views.py]: `get_queryset`

File: foundation/views.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        return super().get_.queryset().select_related(
```
Comment (adamchainz) [inline foundation/views.py]: `get_queryset`

File: djangoproject/templates/foundation/meeting_detail.html [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+<h2>Grants approved </h2>
+
+<ul>
+  {% for grant in meeting.grants_approved.all %}
+  <li>{{ grant }}</li>
+  {% endfor %}
+</ul>
+{% endif %}
+
+{% if meeting.individual_members_approved.exists %}
```
Comment (adamchainz) [inline djangoproject/templates/foundation/meeting_detail.html]: do `.all` then the query won't be repeated in the loop?

File: djangoproject/templates/foundation/meeting_detail.html [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+</ul>
+{% endif %}
+
+{% if ongoing_business %}
+<h2>Ongoing business</h2>
+
+{% for business in ongoing_business %}
+<h3>{{ business.title }}</h2>
+
+{{ business.body|safe }}
```
Comment (adamchainz) [inline djangoproject/templates/foundation/meeting_detail.html]: is it really safe? :)

Comment (ubernostrum) [reply djangoproject/templates/foundation/meeting_detail.html]: Yes -- it's coming from the admin interface where it will be entered by the DSF Secretary.

**Comment by ubernostrum** [comment]
This deliberately does not tie to the existing membership app's models, for two reasons:

1. Thee's an RFP open to completely redo the membership system.
2. This is just about recording and displaying members approved during a particular meeting. If we later want it to tie to the membership system it can, but one of the goals here is to get meeting minutes published quickly, and tying it to the full bureaucracy of the membership system would slow it down.

Regarding tests, I can write some but I'm not really sure what they'd do; they'd mostly be testing that Django's generic views do what Django's documentation says.

And as far as the specific structure: the structure of the DSF meeting minutes hasn't changed in years; this just replicates the structure on djangoproject.com. And honestly, *any* improvement on the current flow is a welcome thing at this point; the meeting minutes are a horrible tedious thing right now, which is why we so often build up multi-month backlogs of unposted minutes. I'd really like to get this merged as soon as possible if the code passes muster.

**Comment by tobiasmcnulty** [comment]
> Regarding tests, I can write some but I'm not really sure what they'd do; they'd mostly be testing that Django's generic views do what Django's documentation says.

A couple ideas:

- Do the archive URLs load and render the expected objects?
- Does the meeting detail URL load and render the expected objects?

`members/test_views.py` has a few tests you could borrow from.

> And as far as the specific structure: the structure of the DSF meeting minutes hasn't changed in years; this just replicates the structure on djangoproject.com.

I looked through past meetings, and honestly I did not see a whole lot of consistency to the structure from year to year. I'm not saying that's a _bad_ thing; I'm more wondering how much effort it will be to fit past meetings into this structure (if that's the plan, which it sounded like it might be) and how future boards and secretaries will adapt (to) this structure / data model...

File: foundation/admin.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+
+@admin.register(models.GrantCurrency)
+class GrantCurrencyAdmin(admin.ModelAdmin):
+    pass
+
+
+class GrantInline(admin.TabularInline):
+    extra = 0
+    min_num = 5
```
Comment (tobiasmcnulty) [inline foundation/admin.py]: Is this intentional (same for `BusinessInline`)? As-is, I can't save a meeting with fewer than 5 grants approved or business items entered...

Comment (ubernostrum) [reply foundation/admin.py]: I've been fiddling with the numbers based on what we've historically had. About to change back to defaults.

**Comment by ubernostrum** [comment]
@tobiasmcnulty Part of my reticence on tests is the number of objects that need to be created just to successfully save a single instance of `Meeting`, and which really would just serve the purpose of "yup, Django's generic views work".

**Comment by glasnt** [comment]
Hi; 2018 DSF Secretary here. 

Practically speaking, this app has exactly what is currently needed to easily transcribe minutes from a standard DSF meeting, while enforcing some form to any adhoc discussions (any other discussions outside of the normal business form into a item of Business, both practically and in this app)

Given the pending implementation of additional functionality for DSF membership, etc, this form shouldn't define any of that, but should instead link into it later. Yes that might not be practical, but it'll work for now, and that's what the DSF Secretary needs.

**Comment by ubernostrum** [comment]
The Board just held its April meeting, and we have a consensus in favor of what this app is doing. Unless there are strong objections, I'd like to merge this before I leave for PyCon.

@carltongibson @felixxm do either of you have technical comments on this?

**Pull request merged**